ipfsconn/ipfshttp/ipfshttp.go

```diff
@@ -918,3 +924,20 @@ func (ipfs *Connector) SwarmPeers() (api.SwarmPeers, error) {
 	}
 	return swarm, nil
 }
+
+// extractCid extracts the cid argument from a url.URL, either via
+// the query string parameters or from the url path itself.
+func extractCid(u *url.URL) (string, bool) {
```

Comment:
rename to `extractArgument`, since the argument doesn't have to be a Cid in all cases (even if now it's always one).